src/dataIntegration/metabotools/calculateLODs.m

```diff
@@ -14,14 +14,30 @@
 %   lod_mM:            Detection limits in mM
 %
 % .. Author: - Maike K. Aurich 27/05/15
+%  Modified by Loic Marx, November 2018
 
-lod_gL = lod_ngmL* 0.000001;
+% input checking
+if length(lod_ngmL) == 1
+    
+    lod_ngmL = repmat(lod_ngmL, 1, length(theo_mass)); 
+    
+    % throw a warning
+    warning('The inputs have different size but we managed to fix this problem');
+    
+end    
 
-lodmM =[];
+if length(lod_ngmL) ~= length(theo_mass)
```

Comment:
what you are looking for here is not `length` but `numel`. 
`[2,2;2,2]` and `[2,1]` have the same length but are still invalid inputs.

Reply:
I prefer `size()`...

Reply:
Yes, but.
Size looks into both dimensions, at which point you have to either make a complicated statement or you only allow input that is either a column or a row vector.